manifests/ca.pp

```diff
@@ -41,79 +42,60 @@
     org_unit      => $org_unit,
     expiration    => $ca_expiration,
     generate      => $generate,
-    deploy        => $deploy,
+    deploy        => false,
     password_file => $ca_key_password_file,
     build_dir     => $certs::ssl_build_dir,
   }
   $default_ca = Ca[$default_ca_name]
 
+
   if $server_cert {
     ca { $server_ca_name:
       ensure        => present,
       generate      => $generate,
-      deploy        => $deploy,
+      deploy        => false,
       custom_pubkey => $certs::server_ca_cert,
       build_dir     => $certs::ssl_build_dir,
     }
   } else {
     ca { $server_ca_name:
-      ensure    => present,
-      generate  => $generate,
-      deploy    => $deploy,
-      ca        => $default_ca,
-      build_dir => $certs::ssl_build_dir,
+      ensure        => present,
+      generate      => $generate,
+      deploy        => false,
+      custom_pubkey => "${certs::ssl_build_dir}/${default_ca_name}.crt",
+      build_dir     => $certs::ssl_build_dir,
     }
   }
-  $server_ca = Ca[$server_ca_name]
 
   if $generate {
     file { "${ssl_build_dir}/KATELLO-TRUSTED-SSL-CERT":
       ensure  => link,
       target  => "${ssl_build_dir}/${server_ca_name}.crt",
-      require => $server_ca,
+      require => Ca[$server_ca_name],
     }
   }
 
   if $deploy {
-    Ca[$default_ca_name] ~>
-    pubkey { $ca_cert:
-      key_pair => $default_ca,
-    } ~>
-    pubkey { $ca_cert_stripped:
-      strip    => true,
-      key_pair => $default_ca,
-    } ~>
-    file { $ca_cert:
+    # Ensure CA key deployed to /etc/pki/katello/private no longer exists
+    # The CA key is not used by anything from this directory and does not need to be deployed
+    file { $ca_key:
+      ensure => absent,
+    }
+
+    file { $certs::katello_default_ca_cert:
       ensure => file,
-      owner  => 'root',
+      source => "${certs::ssl_build_dir}/${default_ca_name}.crt",
+      owner  => $owner,
       group  => $group,
-      mode   => '0644',
+      mode   => '0440',
```

Comment:
Why should this only be visible to the Foreman user? IMHO this should be root:root with mode 0644. Same below with the server CA cert.